src/optimizer_interface.jl

```diff
@@ -73,51 +73,58 @@ function MOIU.attach_optimizer(model::Model)
 end
 
 """
-    set_optimizer(model::Model, optimizer_factory;
-                  bridge_constraints::Bool=true)
-
+    set_optimizer(
+        model::Model,
+        optimizer_factory;
+        bridge_constraints::Bool = false,
+    )
 
 Creates an empty `MathOptInterface.AbstractOptimizer` instance by calling
-`optimizer_factory()` and sets it as the optimizer of `model`. Specifically,
-`optimizer_factory` must be callable with zero arguments and return an empty
-`MathOptInterface.AbstractOptimizer`.
+`MOI.instantiate(optimizer_factory)` and sets it as the optimizer of `model`.
 
-If `bridge_constraints` is true, constraints that are not supported by the
-optimizer are automatically bridged to equivalent supported constraints when
-an appropriate transformation is defined in the `MathOptInterface.Bridges`
-module or is defined in another module and is explicitly added.
+If `bridge_constraints`, adds a `MOI.Bridges.LazyBridgeOptimizer` layer around
+the constructed optimizer.
 
-See [`set_optimizer_attributes`](@ref) and [`set_optimizer_attribute`](@ref) for setting
-solver-specific parameters of the optimizer.
+See [`set_optimizer_attributes`](@ref) and [`set_optimizer_attribute`](@ref) for
+setting solver-specific parameters of the optimizer.
 
 ## Examples
+
 ```julia
 model = Model()
+
 set_optimizer(model, GLPK.Optimizer)
+
+set_optimizer(model, () -> Gurobi.Optimizer(); bridge_constraints = true)
+
+factory = optimizer_with_attributes(Gurobi.Optimizer, "OutputFlag" => 0)
+set_optimizer(model, factory)
 ```
 """
 function set_optimizer(
     model::Model,
     optimizer_constructor;
-    bridge_constraints::Bool = true,
+    bridge_constraints::Bool = false,
```

Comment:
Maybe put `length(model.bridge_types) > 0` here so that the user can still force it to `false` even if bridges are added.